aegis-web/yo/app/views/datasetsBrowser.html

```diff
@@ -52,18 +51,61 @@
+                      &nbsp;Messages
+                    </a>
+                    <ul class="dropdown-menu dropdown-messages" >
+                      <li class="dropdown-header text-muted" ng-if="mainCtrl.messages.length"><h4>Latest messages</h4></li>
+                      <li class="dropdown-header text-muted" ng-if="!mainCtrl.messages.length"><h4>Inbox empty.</h4></li>
+                      <li ng-repeat="msg in mainCtrl.messages| limitTo:5">
+                        <div>
+                          <div ng-click="mainCtrl.openMessageModal(msg)">                    
+                            <div class="message-subject text-muted">
+                              <i ng-if="msg.unread" class="fa fa-circle pull-left" style="color:#f89406; margin-top: 4px;" title="Unread message"></i>
+                              <i ng-if="!msg.unread" class="fa fa-circle pull-left" style="color:grey; margin-top: 4px;" title="Read message"></i>
+                              {{msg.subject}}</div>
+                            <div class="text-muted">{{msg.dateSent| date:'medium'}}</div>
+                          </div>
+                          <div ng-click="mainCtrl.openMessageModal(msg)" class="message-preview">{{msg.preview| limitTo:80}} {{msg.preview.length > 80 ? ' ...' : ''}}</div>
+                          <a ng-if="msg.path" href="#!/{{msg.path}}"> go to requested path.</a>
+                        </div>
+                      </li>
+                      <a class="pull-right" ng-if="mainCtrl.messages.length" ng-click="mainCtrl.openMessageModal()" style="text-decoration:none;"> See all messages</a>
+                      <a class="pull-right" ng-if="!mainCtrl.messages.length" ng-click="mainCtrl.openMessageModal()" style="text-decoration:none;"> Goto messages</a>
+                    </ul>
+                  </li>          
+                </ul>
+              </div>
+            </div>
+          <div class="row content-utility-bar content-utility-bar-actions" ng-if="!mainCtrl.showSearchPage">
               <div class="col-md-3">
                 <h4 style="margin-top: 10px; margin-bottom: 5px"><a ng-click="datasetsCtrl.goToDataSetsDir()">DataSets</a>
                 <span ng-show="datasetsCtrl.pathArray.length > 10">...</span>
                 <span data-ng-repeat="path in datasetsCtrl.pathArray.slice(datasetsCtrl.pathArray.length - 1 - datasetsCtrl.breadcrumbLen()) track by $index">
-                  <span> > </span><a ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</a>
+                  <span> > </span><span ng-click="datasetsCtrl.goToFolder($index + 1)">{{path| limitTo: 16 }}{{path.length > 16 ? '..' : ' '}}</span>
                 </span>
-                <span><a ng-click="datasetsCtrl.goToFolder(datasetsCtrl.pathArray.length)"> <span class="fa fa-refresh"></span></a></span>
+                <span><a ng-click="datasetsCtrl.goToFolder(datasetsCtrl.pathArray.length)"></a></span>
```

Comment:
if you want I put the refresh button that was there before

Reply:
Since this is essentially doing the same as clicking the last path above, I would try to put ` <span class="fa fa-refresh fa-sm"></span>` in the same a tag as the text with an nf-if statement so that it's visible only on the last one. `fa-sm` class should make the icon smaller to match the text size. In any case, remove this line.